app/models/proposal.rb

```diff
@@ -30,12 +30,12 @@ class Proposal < ActiveRecord::Base
   has_many :individual_approvals, ->{ individual }, class_name: 'Approvals::Individual'
   has_many :approvers, through: :individual_approvals, source: :user
   has_many :api_tokens, through: :individual_approvals
-  has_many :attachments
+  has_many :attachments, dependent: :destroy
```

Comment:
We added foreign key constraints in `db/migrate//20150921183423_add_db_constraints.rb` for the same reasons outlined in that article (a nice piece btw). I had intentionally *not* included `ON DELETE CASCADE` to the FK definitions because my assumption was that we did not ever want to delete anything from the db, and that the FK definitions should help make it really hard to delete anything. If we circumvent that db constraint by having AR cascade the deletes at the code level, that undoes some of the data retention guards the db constraint provides.

That said, if we really cared about data retention as a business policy, I think we should be far more intentional about it, rather than abusing a happy side effect of the db constraints. So I would be inclined to (a) not include the `dependent: destroy` and (b) take a more systematic approach to the data retention issue, similar to what we've done with the authorization audit. Story created in https://www.pivotaltracker.com/story/show/106870524